common/src/main/java/org/opensearch/ml/common/ResourceSharingClientAccessor.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common;
+
+import org.opensearch.security.spi.resources.client.ResourceSharingClient;
+
+/**
+ * Accessor for resource sharing client
+ */
+public class ResourceSharingClientAccessor {
+    private ResourceSharingClient CLIENT;
+
+    private static ResourceSharingClientAccessor resourceSharingClientAccessor;
+
+    private ResourceSharingClientAccessor() {}
+
+    public static ResourceSharingClientAccessor getInstance() {
```

Comment:
Do we need to make is thread safe? Or it will be accessing from single thread only ?

Reply:
We'll be following singleton pattern and instance doesn't contain any values that can be mutated post boot-strap. The resource client will be supplied by security plugin on bootstrap.